mayastor/src/subsys/nvmf/admin_cmd.rs

```diff
@@ -134,6 +133,22 @@ extern "C" fn nvmf_create_snapshot_hdlr(req: *mut spdk_nvmf_request) -> i32 {
     }
 }
 
+pub fn create_snapshot(
+    lvol: Lvol,
+    cmd: &spdk_sys::spdk_nvme_cmd,
+    io: *mut spdk_sys::spdk_bdev_io,
+) {
+    let snapshot_time = unsafe {
+        spdk_sys::nvme_cmd_cdw10_get_val(&*cmd) as u64
+            | (spdk_sys::nvme_cmd_cdw11_get_val(&*cmd) as u64) << 32
+    };
+    let snapshot_name = Lvol::format_snapshot_name(&lvol.name(), snapshot_time);
+    // Blobfs operations must be on md_thread
+    Reactors::master().send_future(async move {
+        lvol.create_snapshot_local(io, &snapshot_name).await;
+    });
```

Comment:
do we need to wait for the inner call to complete? Otherwise when we leave `create_snapshot` the snapshot might not be created yet and we don't know if it failed?

Reply:
Also handled with io_completion_local. It's basically simulating the effect of a child IO without actually issuing a child IO (in the usual sense) as it's issuing a vbdev_lvol operation instead.